### DISCUSSION TITLE
Allow button/switch to be used for external/custom control

### DISCUSSION THREAD

**Discussion by daninden** [dingo35/SmartEVSE-3.5#136] [Ideas]
I was looking at making the button/switch of the SmartEVSE available for custom purposes.
As I am on a dynamic energy plan these days I would like to take control of the EVSE (when to charge, and how fast to charge).
But also allow someone to override the settings if immidiate charging is required (for a specific amount of time).

I was thinking of exposing the button to the api/mqtt (get and set) and changing led color to blue when button is activated.
In Home Assistant I can then make decisions on the state of the button.

The question is, would this be something beneficial to this project, as I am not that much interested in maintaining my own version.
I already made some source code changes, but will only start testing it if it might get incoorporated in the released application.

My changes for now: https://github.com/daninden/SmartEVSE-3/pull/1

**Comment by dingo35** [comment]
Hi Dan, we were happy with the code you donated previously, and we're happy to incorporate changes like this.

I just merged a PR which makes it possible to set the led colors by API, so there might be some overlap with your proposal. If you can solve that Im happy to commit your contribution!

**Reply by daninden** [reply]
Great! then I'll go clean/tidy it up and see if it works as expected :)
